app/src/main/java/com/codesoom/assignment/interceptor/LoginCheckInterceptor.java

```diff
@@ -0,0 +1,76 @@
+package com.codesoom.assignment.interceptor;
+
+import com.codesoom.assignment.annotation.CheckJwtToken;
+import com.codesoom.assignment.application.AuthenticationService;
+import com.codesoom.assignment.errors.InvalidTokenException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.handler.AbstractHandlerMapping;
+import org.springframework.web.servlet.resource.ResourceHttpRequestHandler;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Component
+public class LoginCheckInterceptor implements HandlerInterceptor {
+
+    private AuthenticationService authenticationService;
+
+    LoginCheckInterceptor(AuthenticationService authenticationService){
+        this.authenticationService = authenticationService;
+    }
+
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+            throws Exception {
+
+        if(request.getMethod().equals("OPTIONS")) {
+            return true;
+        }
+
+        if (handler instanceof ResourceHttpRequestHandler) {
+            return true;
+        }
+
+
+        try{
+            boolean hasAnnotation = checkAnnotation(handler, CheckJwtToken.class);
+
+            if(hasAnnotation){
+                String accessToken = getAccessToken(request);
+                authenticationService.parseToken(accessToken);
+            }
+        }catch(InvalidTokenException e){
+            request.getRequestDispatcher("/session/error").forward(request, response);
+            return false;
+        }
+        return true;
```

Comment:
try catch 뒤에는 다른 코드가 오지 않는게 좋아요. 이해하기 어렵게 만듭니다.

```suggestion
        try{
            boolean hasAnnotation = checkAnnotation(handler, CheckJwtToken.class);

            if(hasAnnotation){
                String accessToken = getAccessToken(request);
                authenticationService.parseToken(accessToken);
            }
            
            return true;
        }catch(InvalidTokenException e){
            request.getRequestDispatcher("/session/error").forward(request, response);
            return false;
        }
```

Reply:
넵 수정했습니다 감사합니다!